src/transactions.rs

```diff
@@ -39,7 +42,7 @@ pub struct Transaction<L: Ledger> {
     pending_uids: HashSet<u64>,
     /// A receiver memo for each output, except for burned outputs.
     memos: Vec<Option<ReceiverMemo>>,
-    sig: Signature,
+    sig: Option<Signature>,
```

Comment:
Since `memos` and `sig` are related (they should always be both present or neither) does it make sense to do `memos: Option<(Vec<Option<ReceiverMemo>>, Signature)>`? Maybe even create a little struct like `SignedMemos { memos, sig }`